server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java

```diff
@@ -240,6 +240,152 @@ public void getMessageListShouldFilterMessagesWhenMultipleInMailboxesFilterMatch
     }
 
     @Test
+    public void getMessageListShouldFilterMessagesWhenNotInMailboxesFilterMatches() throws Exception {
+        jmapServer.serverProbe().createMailbox(MailboxConstants.USER_NAMESPACE, username, "mailbox");
+        jmapServer.serverProbe().appendMessage(username, new MailboxPath(MailboxConstants.USER_NAMESPACE, username, "mailbox"),
+                new ByteArrayInputStream("Subject: test\r\n\r\ntestmail".getBytes()), new Date(), false, new Flags());
+
+        jmapServer.serverProbe().createMailbox(MailboxConstants.USER_NAMESPACE, username, "mailbox2");
+        await();
+
+        List<String> mailboxIds = 
```

Comment:
If you use the serverProbe to get the mailbox IDs, you will end up with a more concise, robust and readable code.